### PR TITLE
[GPU] Fix priorbox constant propagation

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - Fix priorbox constant propagation
 - Add priorbox kernel validate() to avoid clustered version which causes seg fault
 - Add unit test for dynamic shape case
 - Update existing unit test/func test for new flow

### Tickets:
 - 101235

### CODE REVIEW

File: src/tests/functional/shared_test_classes/src/single_layer/prior_box.cpp [hunk 1]
```diff
@@ -90,8 +90,6 @@ void PriorBoxLayerTest::SetUp() {
         shape_of_2,
         attributes);
 
-    ov::pass::disable_constant_folding(priorBox);
```
Comment (yeonbok) [inline src/tests/functional/shared_test_classes/src/single_layer/prior_box.cpp:93]: As talked offline, this change will be reverted

Comment (wilson-seok) [reply src/tests/functional/shared_test_classes/src/single_layer/prior_box.cpp]: Done. Thanks!

File: src/plugins/intel_gpu/src/kernel_selector/kernels/prior_box/prior_box_kernel_ref.cpp [hunk 2]
```diff
@@ -82,11 +82,15 @@ bool PriorBoxKernelRef::Validate(const Params& params, const optional_params& op
...
-    const auto& priorBox8arams = dynamic_cast<const prior_box_params&>(params);
-    if (priorBox8arams.inputs.size() != 2) {
+    const auto& priorBoxParams = dynamic_cast<const prior_box_params&>(params);
+    if (priorBoxParams.inputs.size() != 2) {
         return false;
     }
 
+    // Current ref kernel doesn't support clustered version
+    if (priorBoxParams.is_clustered) {
+        return false;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/kernel_selector/kernels/prior_box/prior_box_kernel_ref.cpp:92]: If we dont support a spec, then could you add an exception handling at the creation of the op?

File: src/plugins/intel_gpu/tests/unit/dynamic_execution/priorbox_test.cpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                                attrs.offset,
+                                attrs.scale_all_sizes,
+                                attrs.fixed_ratio,
+                                attrs.fixed_size,
+                                attrs.density,
+                                false),
+                      reshape("reshape1", input_info("priorBox1"), input_info("pattern_id1"), false, ov::PartialShape::dynamic(3)),
+                      input_layout("input3", in_layout_1),
+                      input_layout("input4", in_layout_2),
+                      //data("input4", input2_mem),
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/unit/dynamic_execution/priorbox_test.cpp:110]: Please remove unused code

File: src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/prior_box.cpp [hunk 4]
```diff
@@ -105,9 +109,9 @@ class PriorBoxLayerGPUTest : public testing::WithParamInterface<PriorBoxLayerGPU
                 attributes_clustered.widths = {86, 13, 57, 39, 68, 34, 142, 50, 23};
                 attributes_clustered.heights = {44, 10, 30, 19, 94, 32, 61, 53, 17};
                 attributes_clustered.variances = {0.1, 0.1, 0.2, 0.2};
-                attributes_clustered.step = 16;
-                attributes_clustered.step_widths = 0;
-                attributes_clustered.step_heights = 0;
+                attributes_clustered.step = 1;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/prior_box.cpp:112]: why are these values changed?

File: src/plugins/intel_gpu/src/kernel_selector/kernels/prior_box/prior_box_kernel_ref.cpp (hunk 2 above)
Comment (wilson-seok) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/prior_box/prior_box_kernel_ref.cpp:92]: I added the assertion in CreateOp(). Thanks!

File: src/plugins/intel_gpu/tests/unit/dynamic_execution/priorbox_test.cpp (hunk 3 above)
Comment (wilson-seok) [reply src/plugins/intel_gpu/tests/unit/dynamic_execution/priorbox_test.cpp]: Done. Thanks!

File: src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/prior_box.cpp (hunk 4 above)
Comment (wilson-seok) [reply src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/prior_box.cpp]: It's not important. But I rollbacked to the original value. Thanks!

**Pull request merged**